Topic coursecontroller

BenBayotas opened the pull request.

BenBayotas committed:
added controller functions: edit and create

BenBayotas committed:
separated course add and view

separated course.blade to course-index.blade for view

BenBayotas committed:
finished course-index table

BenBayotas committed:
added edit function to course controller

BenBayotas committed:
finished the edit functionalities

BenBayotas committed:
combine all cru(d) to one web page

used javascript to auto fill the forms when updating

BenBayotas committed:
fixed route for course add

BenBayotas committed:
deleted redundant blade templates

all crud have been compressed to one blade file: course-index.blade.php

sininen-blue commented on app/Http/Controllers/testcontroller.php:
remove

BenBayotas replied:
done

sininen-blue commented on database/seeders/DepartmentSeeder.php:
reomve this

BenBayotas replied:
done

sininen-blue commented on resources/views/components/layout.blade.php:
```diff
@@ -1,18 +1,14 @@
-
     </head>
-    <body class="container">
+    <body>
```
needs container class to have margin, not needed but nice

also htmx lmao

BenBayotas replied:
need it to be a one pager

BenBayotas replied:
done

sininen-blue commented on resources/views/course/course-index.blade.php:
```diff
@@ -0,0 +1,171 @@
+
+    <!-- JavaScript to populate the form with course data -->
+    <script>
+        function populateForm(id, department_id, code, acronym, description, major, authority_no, accreditation_id, year_granted, years, slots) {
```
probably unnecessary when switching to react, I think we just keep it though

BenBayotas replied:
ill leave it as it is, but may not be implemented in the future

BenBayotas committed:
resolved  convos

-deleted testcontroller
-deleted deptseeder
-for htmx and json, keep it as it is, but wont be implemented in the future

BenBayotas committed:
restored course.blade.php

BenBayotas committed:
Merge branch 'develop' into topic-coursecontroller
